7_drivers_munge/src/GCM_driver_nc_utils.R

```diff
@@ -3,25 +3,99 @@
+  gcm_dates <- as.POSIXct(unique(gcm_data$time), tz='GMT')
+
+  # Pull vector of unique cell numbers
+  gcm_cells <- unique(gcm_data$cell)
+  gcm_cells_dim_name = "gcm_cell_id"
+
+  # Get WGS84 latitude and longitude of cell centroids
+  cell_coords <- spatial_info %>%
+    mutate(lon = sf::st_coordinates(.)[,1], lat = sf::st_coordinates(.)[,2]) %>%
+    rename(cell = cell_no) %>%
+    sf::st_set_geometry(NULL) %>%
+    arrange(cell) %>%
+    filter(cell %in% gcm_cells)
+
+  # Get vectors of cell centroid latitudes and longitudes
+  cell_centroid_lats <- cell_coords %>% pull(lat)
+  cell_centroid_lons <- cell_coords %>% pull(lon)
+
+  # Set up attributes for NetCDF that are independent of variables
+  data_time_units <- "days since 1970-01-01 00:00:00"
+  data_attributes <- list('title' = global_att)
+  data_attributes <- append(data_attributes, grid_params) # add grid parameters to list of global attributes
+  data_coordvar_long_names <- list(instance = "identifier for GCM grid cell", time = "date",
+                              lat = "WGS84 latitude of GCM grid cell centroid", lon = "WGS84 longitude of GCM grid cell centroid",
+                              alt = "NULL")
+
+  # Pivot data to long format to set up for filtering by variable
+  gcm_data_long <- gcm_data %>%
+    pivot_longer(cols = -c(time, cell), names_to = "variable", values_to = "value") %>%
+    arrange(cell)
+
+  # Loop over data variables to populate NetCDF using ncdfgeom::write_timeseries_dsg()
+  for (variable in vars_info$var_name) {
+    # Filter long-format data to selected variable
+    variable_metadata <- vars_info %>% filter(var_name == variable)
+    var_data <- gcm_data_long %>%
+      filter(variable == !!variable) %>%
+      pivot_wider(id_cols = c("time"), names_from = cell, values_from = value) %>%
+      select(-time) %>%
+      as.data.frame() %>%
+      setNames(gcm_cells)
+
+    # Pull attributes for selected variable
+    var_data_unit <- rep(variable_metadata$units, length(gcm_cells))
+    var_data_prec <- variable_metadata$precision
+    var_data_metadata <- list(name = variable_metadata$var_name, long_name = variable_metadata$longname)
+
+    # Check if nc file already exists, and therefore should be added to,
+    # or if it needs to be created
+    var_add_to_existing <- ifelse(file.exists(nc_file), TRUE, FALSE)
+
+    # write this variable to the netcdf file:
+    # ncdfgeom::write_timeseries_dsg() function documentation:
+    # https://github.com/USGS-R/ncdfgeom/blob/master/R/write_timeseries_dsg.R
+    ncdfgeom::write_timeseries_dsg(nc_file,
+                         instance_names = gcm_cells,
+                         lats = cell_centroid_lats,
+                         lons = cell_centroid_lons,
+                         alts = NA,
+                         times = gcm_dates,
+                         data = var_data,
+                         data_unit = var_data_unit,
+                         data_prec = var_data_prec,
```

Comment:
I see this is coming in as `"float"`. Is that more appropriate than `"double"` for these data?

Reply:
Hmm I can make the switch to `"double"` - I see that's what you and Jared used [here](https://github.com/jread-usgs/lakesurf-data-release/blob/main/src/ncdf_utils.R#L148)

Reply:
ooh this is impacting file size (and build time) quite a bit. Now the netCDFs are 436 mb each, before they were 213 mb each.

Reply:
This build is _still going_. Taking at least 5x as long as before.

Reply:
I think I'm going to backtrack on my comment :) since double probably has an unnecessary level of precision, and if we're further reducing precision at the end of the .nc build (see `nco` comment below), it isn't going to help to use `double` here. `float` seems better all around 👍